server/profile.go

```diff
@@ -91,8 +91,87 @@ func GetProfile(w http.ResponseWriter, r *http.Request) {
 }
 
 func EditProfile(w http.ResponseWriter, r *http.Request) {
+	//Connect to datastore
+	ctx := context.Background()
+	dsClient, err := datastore.NewClient(ctx, Conf.Project)
+	if err != nil {
+		log.Println(err)
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte("datastore error"))
+		return
+	}
+
+	//Retrieve user from datastore
+	vars := mux.Vars(r)
+	userID := vars["id"]
+	if len(userID) == 0 {
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte("no ID provided"))
+		return
+	}
+
+	k := datastore.NameKey("User", userID, nil)
+
+	var user User
+	if err := dsClient.Get(ctx, k, &user); err != nil {
+		log.Println(err)
```

Comment:
Please wrap the error into something that gives more context, for example: log.Printf("Cannot retrieve user from DataStore: %v", err) to make debugging easier.